python/unitytrainers/models.py

```diff
@@ -225,8 +225,15 @@ def create_dc_actor_critic(self, h_size, num_layers):
         self.policy = tf.layers.dense(hidden, self.a_size, activation=None, use_bias=False,
                                       kernel_initializer=c_layers.variance_scaling_initializer(factor=0.01))
 
-        self.all_probs = tf.nn.softmax(self.policy, name="action_probs")
-        output = tf.multinomial(self.policy, 1)
+        self.action_masks = tf.placeholder(shape=[None, self.a_size], dtype=tf.float32, name="action_masks")
```

Comment:
`action_masks` should be a global var

Reply:
Why is that ? Shouldn't it be the same as `policy` ?